### PR TITLE
upgrade lftp from 4.4.16 to 4.7.7

### PR DESCRIPTION
**Pull request by walkerlala**

### CODE REVIEW

**Comment by korli** [comment]
You should consider using branches to base your pull requests: the history of commits is becoming huge.

**Comment by walkerlala** [comment]
@korli Sorry for that. Is it possible to do the rebase on Github?

**Comment by walkerlala** [comment]
@korli Sorry for that. Is it possible to do the rebase on Github?

**Comment by korli** [comment]
@walkerlala you have to "pull --rebase" locally, fix conflicts, then force push.

File: net-ftp/lftp/lftp-4.7.7.recipe [hunk 1]
```diff
@@ -62,19 +63,15 @@ GLOBAL_WRITABLE_FILES="
...
 {
 	libtoolize --force --copy --install
 	aclocal -I m4
 	autoconf
 	automake
-	runConfigure --omit-dirs binDir ./configure --with-openssl --enable-static \
-		--bindir=$prefix/bin
+	runConfigure --omit-dirs binDir ./configure --with-openssl \
+		 --enable-static --bindir=$prefix/bin \
+		CPPFLAGS=-I/boot/system/develop/headers/readline
```
Comment (korli) [inline net-ftp/lftp/lftp-4.7.7.recipe]: This can't work on x86_gcc2h

Comment (walkerlala) [inline net-ftp/lftp/lftp-4.7.7.recipe]: Any error output message?

Comment (korli) [inline net-ftp/lftp/lftp-4.7.7.recipe]: I mean: on x86_gcc2h this path is invalid.
BTW usually using readline headers happen with: 
#include <readline/readline.h>
So if you try to add this path to the include paths, there might be something wrong with the include macro, and you might want to patch this one instead. Hope it helps.

**Comment by walkerlala** [comment]
Rebase done.

As for the x86_gcc2h issue, lftp4.4.16 did not compile either. Why don't we leave this for a while.

**Comment by Begasus** [comment]
did some work on it, maybe you could check this? 
https://github.com/Begasus/haikuports/commit/a972ef151d689019af67961cc90220d994210cac

**Comment by Begasus** [comment]
This one can be closed? 
https://github.com/haikuports/haikuports/pull/1546

**Pull request closed without merging**